src/DropdownMenu.js

```diff
@@ -1,12 +1,99 @@
 import PropTypes from 'prop-types';
-import React, { useContext, useRef, useEffect } from 'react';
-import { Popper } from 'react-popper';
-import usePrevious from '@restart/hooks/usePrevious';
+import { useContext, useRef } from 'react';
 
+import useCallbackRef from '@restart/hooks/useCallbackRef';
+import usePopper from './usePopper';
 import useRootClose from './useRootClose';
-import { useMergedRefs } from './utils/mergeRefs';
 import DropdownContext from './DropdownContext';
 
+export function useDropdownMenu(options = {}) {
+  const context = useContext(DropdownContext);
+
+  const [arrowElement, attachArrowRef] = useCallbackRef();
+
+  const hasShownRef = useRef(false);
+
+  const {
+    flip,
+    rootCloseEvent,
+    popperConfig = {},
+    usePopper: shouldUsePopper = true,
+  } = options;
+
+  const show = context.show == null ? options.show : context.show;
+  const alignEnd =
+    context.alignEnd == null ? options.alignEnd : context.alignEnd;
+
+  if (show && !hasShownRef.current) {
+    hasShownRef.current = true;
+  }
+
+  const handleClose = e => {
+    if (!context.toggle) return;
+
+    context.toggle(false, e);
```

Comment:
```suggestion
    context.toggle?.(false, e);
```

Reply:
Optional chaining plugin is not specified in babel config